### ISSUE TITLE
Allow cusomizing the hotkey that shows/hides inlay hints

### ISSUE THREAD

**Issue by HolyBlackCat** [microsoft/vscode#149019]
Currently the setting `"editor.inlayHints.enabled": "{on,off}UnlessPressed",` forces you to use <kbd>Ctrl</kbd><kbd>Alt</kbd> to show/hide the inlay hints.

I think it would be great to support customizing this hotkey, for example I'd bind one of the F keys to it.

**Comment by wesbos** [comment]
Yes, this would be amazing to have. Currently I have them set to offUnlessPressed, but I sometimes want to turn them on  - because I'm teaching or need a screenshot. 

A command to toggle them on, and then back to off / offUtillPressed would be very helpful.

**Comment by eduarddejong** [comment]
Would be great, because on my system Ctrl+Alt activates the menu bar, places the focus there (I have just enabled the feature, I don't know how normal this is?). Seems kinda buggy to me.
I am working on the Xfce desktop on Linux. I believe not so many people are working with VS Code here or are testing it here.

**Comment by burner1024** [comment]
I've tried this feature for 15 minutes, and my hand already hurts. Ctrl-Alt is an ok combination to press once, but for holding it's terrible.
The losing focus issue also doesn't increase happiness. You can keep focus by using Alt-Ctrl instead of Ctrl-Alt, but this is even more awkward.

**Comment by ObstinateM** [comment]
Hi, I'm still looking forward to this feature! Any updates?

**Comment by daira** [comment]
I don't see why this wasn't a toggle in the first place. Type hints are most useful when you actually have a type error, or want to refactor the types, or are trying to understand someone else's code. Those are activities that take up a low proportion of my time overall, but when I'm doing them I don't want to have to hold down an awkward key combination, which prevents actually modifying the code while seeing the hints!

**Comment by starball5** [comment]
Related on Stack Overflow: [How can I change the inlay hints keyboard shortcut in Visual Studio Code?](https://stackoverflow.com/q/78820208/11107541)

**Comment by onelaview** [comment]
I'm on MacOS and my keyboard workflow heavily relies upon the [HYPER key](https://benholmen.com/blog/hyper-key-with-karabiner-elements-raycast/) using [Karabiner-Elements](https://karabiner-elements.pqrs.org/). And my setup is using `Left Shift + Right Control + Right Option` as the `HYPER` key. In VS Code, I have set the `editor.inlayHints.enabled` to `offUnlessPressed`.

As you can see, because the the show/hide inlay shortcut is also using `Control + Option` so it's somewhat conflicting with my `HYPER` key setup. This means that every time I use my HYPER key, it also triggers the inlay hints causing text in the editor flashing/flickering.

For example, in this video I have `HYPER + I` mapped to `Arrow Up` and `HYPER + K` mapped to `Arrow Down` key for moving cursor. When I move cursor using `HYPER` key, the text in the editor area flashes.


https://github.com/user-attachments/assets/26476e9b-d981-4852-987b-00910fcbaeaa

> Video Recorded from 
> - MacOS Sonoma 14.6.1
> - VS Code Version: 1.93.1 Commit: 38c31bc77e0dd6ae88a4e9cc93428cc27a56ba40

**Comment by Ilanaya** [comment]
The other issue is that `ctrl + alt` for toggling inlay hints forces you to [open the type definition on the side](https://code.visualstudio.com/docs/editing/editingevolved#:~:text=the%20side%20with-,Ctrl%2BAlt%2BClick,-.) 
> Tip: You can jump to the definition with Ctrl+Click or open the definition to the side with Ctrl+Alt+Click.

This is very annoying as I usually want to jump directly to the definition. AFAIK there is currently no way to get the desired behaviour unless I'll use `"editor.inlayHints.enabled": "on" `

**Comment by ShalokShalom** [comment]
I am also experiencing this, and in Zen mode, this shows the top menu bar, breaking the setup. I cannot have `show/hide inline hints` at the same time as toggling the menu bar. That's a clear bug. Thanks :)

**Comment by sromeroi** [comment]
It would be great to have a **Toggle** option so that you don't need to keep the hotkey pressed, and just toggle Inlay Hints on and off.

Also, regarding the current hotkey, when working in VMWare Horizon Client, CTRL+ALT takes the focus out of the Guest O.S., making the hotkey unusable. You have to mouse-click back in the Guest window to focus it again.
